examples/darwin-framework-tool/commands/common/CHIPCommandBridge.mm

```diff
@@ -111,61 +113,111 @@
+                                                                                                 rootCertificate:certificateIssuer.rootCertificate];
+        [params setOperationalCertificateIssuer:certificateIssuer queue:controllerStorageQueue];
+        params.productAttestationAuthorityCertificates = productAttestationAuthorityCertificates;
+
+        __auto_type * controller = [[MTRDeviceController alloc] initWithParameters:params error:&error];
+        VerifyOrReturnError(nil != controller, MTRErrorToCHIPErrorCode(error), ChipLogError(chipTool, "Controller startup failure."));
+        mControllers[identities[i]] = controller;
     }
 
-    ReturnLogErrorOnFailure([gNocSigner createOrLoadKeys:storage]);
+    return CHIP_NO_ERROR;
+}
 
-    ipk = [gNocSigner getIPK];
+CHIP_ERROR CHIPCommandBridge::SetUpStackWithSharedStorage(NSArray<NSData *> * productAttestationAuthorityCertificates)
+{
+    __auto_type * factory = [MTRDeviceControllerFactory sharedInstance];
+    VerifyOrReturnError(nil != factory, CHIP_ERROR_INTERNAL, ChipLogError(chipTool, "Controller factory is nil"));
+
+    auto factoryParams = [[MTRDeviceControllerFactoryParams alloc] initWithStorage:storage];
+    factoryParams.shouldStartServer = YES;
+    factoryParams.otaProviderDelegate = mOTADelegate;
+    factoryParams.productAttestationAuthorityCertificates = productAttestationAuthorityCertificates;
+
+    NSError * error;
+    auto started = [factory startControllerFactory:factoryParams error:&error];
+    VerifyOrReturnError(started, MTRErrorToCHIPErrorCode(error), ChipLogError(chipTool, "Controller factory startup failed"));
+
+    __auto_type * certificateIssuer = [CertificateIssuer sharedInstance];
 
     constexpr const char * identities[] = { kIdentityAlpha, kIdentityBeta, kIdentityGamma };
     std::string commissionerName = mCommissionerName.HasValue() ? mCommissionerName.Value() : kIdentityAlpha;
     for (size_t i = 0; i < ArraySize(identities); ++i) {
-        auto controllerParams = [[MTRDeviceControllerStartupParams alloc] initWithIPK:ipk fabricID:@(i + 1) nocSigner:gNocSigner];
+        __auto_type * fabricId = @(i + 1);
+        NSNumber * nodeId = nil;
 
+        __auto_type * params = [[MTRDeviceControllerStartupParams alloc] initWithIPK:certificateIssuer.ipk
+                                                                            fabricID:fabricId
+                                                                           nocSigner:certificateIssuer.signingKey];
         if (commissionerName.compare(identities[i]) == 0 && mCommissionerNodeId.HasValue()) {
-            controllerParams.nodeId = @(mCommissionerNodeId.Value());
+            params.nodeId = @(mCommissionerNodeId.Value());
         }
-        // We're not sure whether we're creating a new fabric or using an
-        // existing one, so just try both.
-        auto controller = [factory createControllerOnExistingFabric:controllerParams error:&error];
+
+        params.nodeId = nodeId;
```

Comment:
This looks wrong; it overrides mCommissionerNodeId above, no?

Reply:
> This looks wrong; it overrides mCommissionerNodeId above, no?
You’re absolutely right. This does override mCommissionerNodeId from above. It’s a leftover from previous changes.